### DISCUSSION TITLE
dew heater doesn't  turn on

### DISCUSSION THREAD

**Discussion by Aripaule** [aaronwmorris/indi-allsky#1692] [Q&A] [answered]
hi,

For some reason my dew heater isn't turning on. 

![a](https://github.com/user-attachments/assets/8eb4f564-b79d-4fbc-ae48-268819f34a43)

Any ideas how these settings might affect the on status?

I have a bme280 sensor that I would like to use as a base to turn on the dew heater according to the dew point. 

The dew heater was connected to a relay as shown the diagram below.
![dew](https://github.com/user-attachments/assets/721f09f9-748b-4be2-80cd-2cd4421fcfe6)

temperature and dew point:
![b](https://github.com/user-attachments/assets/e2954ae1-d0c3-4327-9a20-79c1ead462de)

dew heater configuration:
![c](https://github.com/user-attachments/assets/be24e4dc-a00b-4d96-b6ed-14f017daf825)

**Comment by aaronwmorris** [comment]
If you turn off the thresholds for the dew heater, does the dew heater turn on?

**Reply by Aripaule** [reply]
![image](https://github.com/user-attachments/assets/50d6a64f-d0fe-486c-8201-84ac5e566f83)
it stills off

**Reply by aaronwmorris** [reply]
Make sure you reload/restart indi-allsky when you make changes.

Can you attach the output from `grep Sensor /var/log/indi-allsky/indi-allsky.log` ?

**Reply by Aripaule** [reply]
sure. here it goes.

[log. dh.txt](https://github.com/user-attachments/files/17970936/log.dh.txt)

**Reply by aaronwmorris** [reply]
Okay, I think the problem is the user running the program is not a member of the gpio group.  You can manually add the user to the gpio group with the following command:

```
sudo usermod -a -G gpio USERNAME
```

Reboot when finished.

Merged #1693 so that this action is taken care of by default during setup.  You can also just pull down the latest code and re-run setup.sh if you are not comfortable running this manually.

**Reply by Aripaule** [reply]
[log dh.2.txt](https://github.com/user-attachments/files/18000154/log.dh.2.txt)


just with the command "sudo usermod -a -G gpio ari" nothing changed. I should reinstall it from beggining?

**Reply by aaronwmorris** [reply]
I think we fixed the previous issue, but this is a new one.

**Reply by Aripaule** [reply]
damn, any idea?

**Reply by aaronwmorris** [reply]
Your current problem appears to be your TSL2591.  The other sensors appear to be working.  The error you are receiving could be the result of poor wiring.  Double check your connections and see if something is loose.

**Reply by Aripaule** [reply]
Do you think this TSL error could interfere with turning on the dew heater?

**Reply by aaronwmorris** [reply]
Yes.  The hardware sensors and devices (dew heater, fan, etc) are all managed in the same thread.  The sensors are initialized before the dew heater.

I have merged #1697 which should bypass the sensor if this exception occurs again.  Please let me know if this helps.

**Reply by Aripaule** [reply]
Got it. I'll try tomorrow. 

Do you think this could be the cause of failed connection of gps module too?

**Reply by aaronwmorris** [reply]
No, GPS is completely separate from this.

**Reply by Aripaule** [reply]
> Yes. The hardware sensors and devices (dew heater, fan, etc) are all managed in the same thread. The sensors are initialized before the dew heater.
> 
> I have merged #1697 which should bypass the sensor if this exception occurs again. Please let me know if this helps.


Sorry, I'm new to this whole programming thing. To test this new configuration, do I need to do a clean installation, or just replace the file?

**Reply by aaronwmorris** [reply]
You just need to pull down the latest code.

https://github.com/aaronwmorris/indi-allsky/wiki/Updating-indi-allsky

**Reply by Aripaule** [reply]
> You just need to pull down the latest code.
> 
> https://github.com/aaronwmorris/indi-allsky/wiki/Updating-indi-allsky


OK. I updated the software, and restarted but there doesn't seem to be any progress.

[dh3.txt](https://github.com/user-attachments/files/18031365/dh3.txt)

**Reply by aaronwmorris** [reply, marked as answer]
I know it does not seem like it, but we are making progress.  This is a new error.  Lets try switching to a different python module for gpio access.

Try the following:

```
source virtualenv/indi-allsky/bin/activate

pip uninstall RPi.GPIO rpi.lgpio

pip install --upgrade rpi.lgpio
```

**Reply by Aripaule** [reply]
IT WORKEDDDDDDDDDDDDDDDDD!

[dh 4.txt](https://github.com/user-attachments/files/18044257/dh.4.txt)


![image](https://github.com/user-attachments/assets/d6c28274-ef93-47a5-a369-351667fab019)


Now I'm going to test how the limits behave. Could you briefly explain how these limits work? (maybe it's better putting it on the dewheater page)

https://github.com/aaronwmorris/indi-allsky/wiki/Dew-Heater-Support


attached I left the log after the dew heater worked. Can you still identify the problem with the TSL sensor? it was giving strange readings btw

THANK YOU SO FAR

**Reply by aaronwmorris** [reply]
Fantastic!

The limits for the dew heater are designed to work around the dew point.  The temperature is normally higher than the dew point, so the thresholds are above the target.  So if the dew point is 20c a LOW temperature threshold of 15 will mean that when the temperature is 15 degrees above the dew point, 35c, the LOW value kicks in.

Because you are running a standard dew heater, it only supports On and Off, you dew heater would turn on at full power at 35c.  You cannot use multiple levels.

You can configure whatever threshold you want, and even define a manual target if you want to use a temperature only (non dew point) target.

If you want to use a threshold, you need to set the Default level to 0.

**Comment by viktorschmitzarefjew** [comment]
Hello, I have a similar setup. The BME280 is recognized and sending data, however, the dew heater is constantly on via the relay (On/Off), even when the function is set to off. I've already tried different GPIO pins; currently, GPIO 24 is connected. What else could be causing this? 
[indi-allsky_support_info_20251227_22035.txt.gz](https://github.com/user-attachments/files/24356759/indi-allsky_support_info_20251227_22035.txt.gz)
viele grüße
Viktor
